android/src/com/mapswithme/maps/widget/placepage/SponsoredHotel.java

```diff
@@ -73,6 +173,15 @@ static void requestPrice(String id, String currencyCode)
     nativeRequestPrice(id, currencyCode);
   }
 
+  static void requestInfo(String id, String locale)
+  {
+    HotelInfo info = sInfoCache.get(id);
+    if (info != null)
+      onInfoReceived(id, info);
+
+    nativeRequestInfo(id, locale);
```

Comment:
Please explain why you request info even if it's cached and why you call ```onInfoReceived``` only on the cached infos.